pkg/server/handlers.go

```diff
@@ -62,14 +69,62 @@ func getEnvironment(w http.ResponseWriter, r *http.Request) {
 func patchEnvironment(w http.ResponseWriter, r *http.Request) {
 	pathVars := mux.Vars(r)
 
-	err := updateEnvironmentHosts(pathVars["org"], pathVars["env"], r.Header.Get("Authorization"))
+	ns, err := clientset.Core().Namespaces().Get(pathVars["org"] + "-" + pathVars["env"])
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
-		helper.LogError.Printf("Error syncing Hosts to the Environment: %v\n", err)
+		helper.LogError.Printf("Error getting existing Environment: %v\n", err)
 		return
 	}
 
-	w.WriteHeader(http.StatusNoContent)
+	getSecret, err := clientset.Core().Secrets(pathVars["org"] + "-" + pathVars["env"]).Get("routing")
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		helper.LogError.Printf("Error getting existing Secret: %v\n", err)
+		return
+	}
+
+	if os.Getenv("DEPLOY_STATE") == "PROD" {
+		apigeeClient := apigee.Client{Token: r.Header.Get("Authorization")}
+		hosts, err := apigeeClient.Hosts(pathVars["org"], pathVars["env"])
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			helper.LogError.Printf("Error getting Hosts from Apigee: %v\n", err)
+			return
+		}
+
+		ns.ObjectMeta.Annotations["edge/hosts"] = composeHostsJSON(hosts)
+		ns, err = clientset.Core().Namespaces().Update(ns)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			helper.LogError.Printf("Error updating Environment: %v\n", err)
+			return
+		}
+	}
+
+	var jsResponse environmentResponse
+	jsResponse.Name = ns.Name
+	jsResponse.PrivateSecret = getSecret.Data["private-api-key"]
+	jsResponse.PublicSecret = getSecret.Data["public-api-key"]
+
+	//TODO: Parse the annotation and return array of strings
+	tempHosts, err := parseHoststoMap(ns.Annotations["edge/hosts"])
+	if err != nil {
+		errorMessage := fmt.Sprintf("Error Parsing Hosts: %v\n", err)
+		http.Error(w, errorMessage, http.StatusInternalServerError)
+		helper.LogError.Printf(errorMessage)
```

Comment:
That's what I was asking.  I wanted to know if we needed to return or if this was by design and we _could_ handle this gracefully below.